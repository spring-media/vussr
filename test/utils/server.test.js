const { listenAsPromised, closeAsPromised } = require('../../src/utils/server');

function setup() {
  const port = 'port';
  const host = 'host';
  const close = jest.fn();
  const listener = { close };
  const listen = jest.fn().mockImplementation((port, host, cb) => {
    setTimeout(() => cb(null), 0);
    return listener;
  });
  const server = { listen };
  return { listen, close, server, listener, port, host };
}

test('listenAsPromised resolves the listener', async () => {
  const { server, listener, port, host } = setup();
  const returnedListener = await listenAsPromised(server, port, host);
  expect(returnedListener).toBe(listener);
  expect(server.listen).toHaveBeenCalledWith(port, host, expect.any(Function));
});

test('listenAsPromised rejects with an error', async () => {
  const { server, listener, port, host } = setup();
  const error = new Error('Expected error');
  server.listen = jest.fn().mockImplementation((port, host, cb) => cb(error));
  expect(listenAsPromised(server, port, host)).rejects.toThrowErrorMatchingSnapshot();
  expect(server.listen).toHaveBeenCalledWith(port, host, expect.any(Function));
});

test('closeAsPromised waits for the server to be closed', async () => {});

test('closeAsPromised rejects with an error', async () => {});
