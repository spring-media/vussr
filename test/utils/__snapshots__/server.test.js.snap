// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`listenAsPromised rejects with an error 1`] = `"Expected error"`;
